ubuntu/14.04/standard/Dockerfile

```diff
@@ -149,3 +154,21 @@ RUN apt-get update \
  && apt-get install -y --no-install-recommends \
     subversion \
  && rm -rf /var/lib/apt/lists/*
+
+# Download hosted tool cache
+ENV AGENT_TOOLSDIRECTORY=/opt/hostedtoolcache
+RUN azcopy --recursive --source https://vstsagenttools.blob.core.windows.net/tools/hostedtoolcache/linux --destination $AGENT_TOOLSDIRECTORY
+
+# Install the tools from the hosted tool cache
+RUN cd $AGENT_TOOLSDIRECTORY \
+ && for tool in $(ls); do \
+        cd $tool; \
+        for version in $(ls); do \
+            cd $version/x64; \
+            chmod +x setup.sh; \
```

Comment:
Each tool should have a `setup.sh` that runs any installation for the tool.  Ideally these should always be precompiled and need very little setup.  But on Linux at minimum this script will need to mark the tool binaries as executable.